### ISSUE TITLE
HomeConnect API unexpected values (WAV28G43/04)

### ISSUE THREAD

**Issue by smhex** [thoukydides/homebridge-homeconnect#210]
### Plugin Version

0.37.6

### Home Connect Appliance(s)

Bosch Washer WAV28G43/04

### Log File

```shell
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect] ====================================================================================
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect] 
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect] // Option
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect] export interface OptionValues {
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'BSH.Common.Option.EnergyForecast'?:                   number;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'BSH.Common.Option.EstimatedTotalProgramTime'?:        number;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'BSH.Common.Option.FinishInRelative'?:                 number;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'BSH.Common.Option.ProgramProgress'?:                  number;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'BSH.Common.Option.RemainingProgramTime'?:             number;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'BSH.Common.Option.RemainingProgramTimeIsEstimated'?:  boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'BSH.Common.Option.StartInRelative'?:                  number;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'BSH.Common.Option.WaterForecast'?:                    number;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'Dishcare.Dishwasher.Option.ExtraDry'?:                boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'Dishcare.Dishwasher.Option.HygienePlus'?:             boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'Dishcare.Dishwasher.Option.IntensivZone'?:            boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'Dishcare.Dishwasher.Option.SilenceOnDemand'?:         boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'Dishcare.Dishwasher.Option.VarioSpeedPlus'?:          boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Common.Option.LoadRecommendation'?:       number;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Common.Option.ReferToProgram'?:           number; // (unrecognised)
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.IntensivePlus'?:            boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.LessIroning'?:              boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.Prewash'?:                  boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.RinseHold'?:                boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.RinsePlus'?:                RinsePlus;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.SilentWash'?:               boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.Soak'?:                     boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.SpeedPerfect'?:             boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.SpinSpeed'?:                SpinSpeed;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.Stains'?:                   Stains;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.Temperature'?:              Temperature;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect]     'LaundryCare.Washer.Option.WaterPlus'?:                boolean;
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect] }
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect] 
[1/4/2024, 10:44:43 AM] [homebridge-homeconnect] ====================================================================================
```

### Additional Information

I hope that helps. Many thanks for developing this plugin!

**Comment by thoukydides** [comment]
Thanks for the report.

This is a bit odd. I originally added the `LaundryCare.Common.Option.ReferToProgram` program option as taking a number in [v0.35.0](https://github.com/thoukydides/homebridge-homeconnect/compare/v0.34.1...v0.35.0) based on a report in [SmartApfel.Community](https://forum.smartapfel.de/forum/thread/4569-homebridge-homeconnect/?postID=147550#post147550). However, I changed it to taking a `ProgramKey` based on #199 (and supported by #72). However, it really does look like it can be either...

I am not aware of any other Home Connect API options/settings/states/events that can take two different types, so I suspect that someone made a mistake. Anyway, I will change this to `ProgramKey | number` in the next release.

**Comment by thoukydides** [comment]
I have fixed this in [v0.37.8](https://github.com/thoukydides/homebridge-homeconnect/releases/tag/v0.37.8).
Thanks again for the report.

**Comment by thoukydides** [comment]
Just spotted that `LaundryCare.Washer.Option.IntensivePlus` is also listed as a program option in that log, but is currently missing from the plugin. I will include that in the next release.

**Comment by thoukydides** [comment]
Should be fixed in [v0.37.9](https://github.com/thoukydides/homebridge-homeconnect/releases/tag/v0.37.9).
